### PR TITLE
Http compression

### PR DESCRIPTION
**Pull request by ngderek24**

### CODE REVIEW

File: status_counter_test.cc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #include "gtest/gtest.h"
 #include "gmock/gmock.h"
-#include "status_counter.h
```
Comment (jasony0206) [inline status_counter_test.cc]: LOL I know we weren't really testing status counter... but to realize it was missing a " is something else.

**Comment by jasony0206** [comment]
LGTM

File: .gitignore [hunk 2]
```diff
@@ -8,3 +8,5 @@ server
 *.gcno
 *.gcov
 *.dSYM/
+*.pem
```
Comment (alexmwaz) [inline .gitignore]: this is smart hahahaha

File: Makefile [hunk 3]
```diff
@@ -136,6 +143,7 @@ coverage : test
 	gcov -r echo_handler.cc; gcov -r request_handler.cc; \
 	gcov -r static_file_handler.cc; gcov -r status_handler.cc; \
 	gcov -r not_found_handler.cc; gcov -r status_counter.cc;
+	gcov -r compressor.cc;
```
Comment (alexmwaz) [inline Makefile]: would be nice to add coverage for these too
```c++
gcov -r redirect_handler.cc; gcov -r reverse_proxy_handler.cc; \
```

File: Makefile [hunk 4]
```diff
@@ -136,6 +143,7 @@ coverage : test
 	gcov -r echo_handler.cc; gcov -r request_handler.cc; \
 	gcov -r static_file_handler.cc; gcov -r status_handler.cc; \
 	gcov -r not_found_handler.cc; gcov -r status_counter.cc;
```
Comment (alexmwaz) [inline Makefile]: i think there should be a \ at the end of this line

File: status_counter_test.cc (hunk 1 above)
Comment (alexmwaz) [inline status_counter_test.cc]: LOL oh man....

**Review by alexmwaz — COMMENTED** [review]
i think gcov is missing some stuff but looks good!

**Comment by tincan24** [comment]
Addressed issues. Will merge once travis tests pass

**Pull request merged**